### ISSUE TITLE
write [vtk] meshes as binary files, not as plain text

### ISSUE THREAD

**Issue by cchriste** [SCIInstitute/ShapeWorks#1398]
We give away at least 10 significant digits when we perform computations on a mesh and then save the results as a plain text vtk file. The results are abysmal even for computing the average of just two numbers, reducing accuracy down to only two significant digits.

As an example: computing mean normals between two meshes, m03 and m04.
 1. first generating those normals and then computing the average normals
 2. computing the normals, then saving these to files, then reloading them and computing the mean normals

By default, the comparison functions verify accuracy down to five significant figures (a number that can typically be specified, though for our mesh fields is currently hard-coded). For the latter case to succeed, it needs to be changed to only compare down to two significant figures.

**Comment by cchriste** [comment]
Examples of modern vtk file writing: https://kitware.github.io/vtk-examples/site/VTKFileFormats/
Essentially, _we can just set the file type to binary._ Even in this page it is strongly suggested to use more common and broadly supported formats, but at the very least we should be writing binary files.

**Comment by archanasri** [comment]
A fix for this is to use `SetFileTypeToBinary()` when writing vtk files

**Comment by akenmorris** [comment]
I agree that we should probably default to binary (or VTP instead).  I'm not sure what the "more common and broadly supported formats" are.  These are presumably non-VTK formats?  (STL, PLY?)  Of the VTK formats, ASCII legacy VTK is by far the most common and broadly supported (e.g. readable in other tools).  I would check if files written using `SetFileTypeToBinary` are still readable in other tools like CloudCompare.

I don't think the situation for ASCII VTK is as dire as you present it.  I think your problem with having to compare only 2 significant digits is probably due to rounding error somewhere or by converting to decimal for ASCII and back rather than losing "10 significant digits".  It seems the scalar fields from most VTK filters (e.g. normals) are only floats and 6 ASCII digits (%g) is fairly standard.  Are you sure a double precision field is not written out with more than 6 digits?

https://github.com/Kitware/VTK/blob/0bda4b1cb777b25f7b09707bc4cf540cb04cc3c1/IO/Legacy/vtkDataWriter.cxx#L1312

It appears float will be written with %g and double as %11lg

**Comment by cchriste** [comment]
#1399 writes binary as default, putting this issue to QA status
#1400 asks to open this as an optional parameters to all APIs
